spec/integration/client/client_spec.rb

```diff
@@ -60,6 +60,23 @@ def stop_tiny_server
       result.error!
     end
 
+    it "should complete successfully with no other environment variables" do
+      file 'config/client.rb', <<EOM
+local_mode true
+cookbook_path "#{path_to('cookbooks')}"
+EOM
+
+      critical_env_vars = ""
+      %w(PATH RUBYOPT BUNDLE_GEMFILE).each do |option|
+        critical_env_vars += "#{option}=#{ENV[option]} "
+      end
+
+      Chef::Log.info "Bare invocation will have the following load-path."
+      Chef::Log.info shell_out("env -i #{critical_env_vars} ruby -e 'puts $:'").stdout
+      result = shell_out("env -i #{critical_env_vars} #{chef_client} -c \"#{path_to('config/client.rb')}\" -o 'x::default'", :cwd => chef_dir)
+      result.error!
```

Comment:
I think most of them use the implicit test of error! because there's really not much else to verify other than "did this command succeed".  If it didn't, you'd want to see the error log printed (which error! does).  If it did, running a second expect on the return code is redundant because error! already does that.